### PR TITLE
Fix nll loss

### LINKED ISSUE #924: nll_loss dtype
```
import torch
import torch_xla
import torch.nn as nn
import pdb

device = 'xla:0'
m = nn.NLLLoss(None, None, 2).double()
input = torch.rand(15, 10).log().double().to(device)
input.requires_grad = True

target = torch.Tensor(15).uniform_().mul(10).floor().long()
out = m(input, target)
out.sum().backward()

```
Complains about `total_weight` is float.

### PR DESCRIPTION
**Pull request by ailzhang**
Fixes #924 . 
Also fixes edge cases when batch=0 and updated the testcase for better coverage.

### CODE REVIEW

File: torch_xla/csrc/nll_loss.cpp [hunk 1]
```diff
@@ -103,7 +103,9 @@ xla::XlaOp BuildNllLoss(const xla::XlaOp& logits, const xla::XlaOp& labels,
       ValidLabelsCount(labels, ignore_index), logits_shape.element_type());
   xla::XlaComputation add_func =
       XlaHelpers::CreateAddComputation(logits_shape.element_type());
-  return xla::ReduceAll(mul, zero, add_func) / batch;
+  return xla::Select(xla::Ne(batch, zero),
```
Comment (dlibenzi) [inline torch_xla/csrc/nll_loss.cpp]: Since both branches of a Select are executed, it is better to:
```
batch = Select(Ne(batch, 0), batch, 1.0)
```

And then unconditionally divide.

File: torch_xla/csrc/nll_loss.cpp [hunk 2]
```diff
@@ -125,7 +127,10 @@ xla::XlaOp BuildNllLossBackward(const xla::XlaOp& logits,
   xla::XlaOp batch = xla::ConvertElementType(
       ValidLabelsCount(labels, ignore_index), logits_shape.element_type());
   // Compute -one_hot_labels / batch.
-  return xla::Neg(one_hot_labels) / batch;
+  xla::XlaOp zero =
+      XlaHelpers::ScalarValue<float>(0, logits_shape.element_type(), builder);
+  return xla::Select(xla::Ne(batch, zero), xla::Neg(one_hot_labels) / batch,
```
Comment (dlibenzi) [inline torch_xla/csrc/nll_loss.cpp]: Same here.
